core/src/com/group66/game/cannon/ballgraph/BallGraphBreadthFirstConditionalIterator.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * 
+ */
+package com.group66.game.cannon.ballgraph;
+
+import java.util.Iterator;
+
+import com.group66.game.cannon.Ball;
+
+public interface BallGraphBreadthFirstConditionalIterator extends Iterator<Ball> {
+
+
+    /* (non-Javadoc)
+     * @see java.util.Iterator#hasNext()
+     */
+    @Override
+    boolean hasNext();
```

Comment:
For stuff like hasNext, next and size the code is the same in the implementations, maybe make it an abstract class and implement those here? That way you prevent code duplication.